sleuth/sleuth/enrich_tls.py

```diff
@@ -89,9 +102,102 @@ def check_compliance(compliance_policy, scs):
+        elif attr == "sig_alg":
+            param_rules = policy.rules[attr][value]["sig_key_size"]
+
+        if 'operator' in param_rules:
+            op = OPS[param_rules['operator']]
+            del param_rules['operator']
+        else:
+            op = OPS['default']
+
+        # now check each value in the policy using the operator
+        # to ensure that the best fit seclevel is chosen
+        for each in param_rules:
+            if op(param, int(each)):
+                temp_seclevel = param_rules[each]
+                if not current_seclevel or op(temp_seclevel, current_seclevel):
+                    current_seclevel = temp_seclevel
+
+        # if we make it through each value in the policy attribute
+        # and compare with the defined operator, but seclevel
+        # is still undefined, the number is more stringent
+        # than the policy defines, so is assigned max_seclevel
+        if not current_seclevel:
+            current_seclevel = policy.max_seclevel
+
+    if not current_seclevel:
+            current_seclevel = policy.rules[attr][value]["default"]
+
+    return current_seclevel
+
+
+
+# This function analyzes the minimum security level for the certs found in the tls flow
+# based on the supplied policy
+#
+def get_certs_seclevel(policy, certs, unknowns):
+    certs_seclevels = []
+    concerns = []
+
+    if certs:
+        for cert in certs:
+            current_seclevel = UNKNOWN
+
+            if 'sig_alg' in policy.rules:
+                sig_rules = policy.rules['sig_alg']
+
+                if cert['sig_alg'] in sig_rules:
+                    current_seclevel = get_seclevel_with_param(policy, 'sig_alg', cert['sig_alg'], cert['sig_key_size'])
+
+            if current_seclevel <= policy.failure_threshold:
+                concern_string = "sig_alg: " + cert['sig_alg'] + ", sig_key_size: " + str(cert['sig_key_size'])
+                concerns.append(concern_string)
+
+            certs_seclevels.append(current_seclevel)
+
+        # fetch min seclevel element based on whether or not 'unknown' elements should be reported.
+        # default here is 'report'
+        if unknowns == "ignore":
+            certs_seclevel_floor = min([x for x in certs_seclevels if x != UNKNOWN])
+        else:
+            certs_seclevel_floor = min(certs_seclevels)
+
+        ca_result = audit_certs_issuer(certs, policy.trusted_ca_list)
+        if ca_result:
+            concerns.append(ca_result)
+            
+    else:
+        certs_seclevel_floor = None
+        concerns.append('no certs data')
 
+    return certs_seclevel_floor, concerns
 
-def tls_seclevel(policy, unknowns, scs, client_key_length, certs):
+    
+# This function grabs the scs from the reference file and analyzes the minimum security level
+# for the selected cipher suite of the flows based on the supplied policy
+#
+def get_tls_seclevel(policy, scs, client_key_length, unknowns):
```

Comment:
changing name to something like get_scs_seclevel would fit the description of this function better imo